src/core/server/integration_tests/saved_objects/migrations/group3/dot_kibana_split.test.ts

```diff
@@ -364,22 +372,101 @@ describe('split .kibana index into multiple system indices', () => {
         expect(logs).toMatch(`[${index}] Migration completed`);
       });
     });
-  });
 
-  afterEach(async () => {
-    // we run the migrator again to ensure that the next time state is loaded everything still works as expected
-    const { runMigrations } = await migratorTestKitFactory();
-    await clearLog();
-    await runMigrations();
+    afterEach(async () => {
+      // we run the migrator again to ensure that the next time state is loaded everything still works as expected
+      const { runMigrations } = await migratorTestKitFactory();
+      await clearLog();
+      await runMigrations();
 
-    const logs = await readLog();
-    expect(logs).not.toMatch('REINDEX');
-    expect(logs).not.toMatch('CREATE');
-    expect(logs).not.toMatch('UPDATE_TARGET_MAPPINGS');
+      const logs = await readLog();
+      expect(logs).not.toMatch('REINDEX');
+      expect(logs).not.toMatch('CREATE');
+      expect(logs).not.toMatch('UPDATE_TARGET_MAPPINGS');
+    });
+
+    afterAll(async () => {
+      await esServer?.stop();
+      await delay(2);
+    });
   });
 
-  afterAll(async () => {
-    await esServer?.stop();
-    await delay(10);
+  const getAggregatedTypesCountAllIndices = async (esClient: ElasticsearchClient) => {
+    const typeBreakdown = await Promise.all(
+      ALL_SAVED_OBJECT_INDICES.map((index) => getAggregatedTypesCount(esClient, index))
+    );
+
+    return ALL_SAVED_OBJECT_INDICES.reduce<Record<string, Record<string, number> | undefined>>(
+      (acc, index, pos) => {
+        acc[index] = typeBreakdown[pos];
+        return acc;
+      },
+      {}
+    );
+  };
+
+  describe('when multiple Kibana migrators run in parallel', () => {
+    it.each([
+      '7.13.0_concurrent_5k_foo.zip',
+      '7.13.0_with_unknown_so.zip',
+      '7.13.2_so_with_multiple_namespaces.zip',
+      '7.13_1.5k_failed_action_tasks.zip',
+      '7.14.0_xpack_sample_saved_objects.zip',
+      '7.3.0_xpack_sample_saved_objects.zip',
+      '7.7.2_xpack_100k_obj.zip',
+      '7_13_corrupt_and_transform_failures_docs.zip',
+      '8.0.0_document_migration_failure.zip',
+      // '8.0.0_migrated_with_corrupt_outdated_docs.zip' // corrupt package, skipped
+      '8.0.0_migrated_with_outdated_docs.zip',
+      '8.0.0_v1_migrations_sample_data_saved_objects.zip',
+      '8.4.0_with_sample_data_logs.zip',
+    ])('correctly migrates %s archive', async (archive) => {
```

Comment:
We are going to make the CI group run time explode with this, as it's represents the time equivalent of 10+ tests. If you want to run all those archives though the suite, I'd say you will have to move this test into it's dedicated test group.

Also, I'm not sure how relevant / necessary it is to run the test against that many archives tbh. It feels like you just did `select *.zip` instead of picking the relevant one?

Reply:
Yep, I thought about creating a dedicated group just for the split, I don't know the impact though.

TBH I did this as part of the 8.8.0 test spike, I wanted to exhaustively test all of them to see if there were any failures. Turns out only the `8.0.0_migrated_with_corrupt_outdated_docs.zip` failed, as the archive is in bad shape (expired license; the [related test](https://github.com/elastic/kibana/blob/main/src/core/server/integration_tests/saved_objects/migrations/group2/corrupt_outdated_docs.test.ts#L47) is _skipped_ BTW).

I also took a look at the logs and the `.snap` file (I dumped a breakdown of **_indices => types_** before and after). Good news is that all of the migrators' logs look fine, and the breakdown makes total sense.

That being said, I don't know if it's worth merging the PR. Or perhaps it's not worth adding all ZIP files, I'll select 2 or 3 representative ones.

Reply:
I would be fine merging with 2 or 3 archives, I think the test makes sense overall (even If I'm a bit scared of the flakiness running integration tests with 6 'instances', but we'll see in the long run)